plugins/modules/ec2_vpc_dhcp_option.py

```diff
@@ -190,91 +190,256 @@
+    This is the only place we use it, so create it now
+    https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateDhcpOptions.html
+    """
+    new_config = []
+    if module.params['dns_servers'] is not None:
+        new_config.append({'Key': 'domain-name-servers', 'Values': module.params['dns_servers']})
+    if module.params['netbios_name_servers'] is not None:
+        new_config.append({'Key': 'netbios-name-servers', 'Values': module.params['netbios_name_servers']})
+    if module.params['ntp_servers'] is not None:
+        new_config.append({'Key': 'ntp-servers', 'Values': module.params['ntp_servers']})
+    if module.params['domain_name'] is not None:
+        new_config.append({'Key': 'domain-name', 'Values': [module.params['domain_name']]})
+    if module.params['netbios_node_type'] is not None:
+        new_config.append({'Key': 'netbios-node-type', 'Values': module.params['netbios_node_type']})
+        # if module.params[option] is not None:
+        #     create_dhcp_options.append({'Key': option, 'Value': module.params[option]})
+    try:
+        if not module.check_mode:
+            dhcp_options = client.create_dhcp_options(aws_retry=True, DhcpConfigurations=new_config)
+            return changed, dhcp_options['DhcpOptions']['DhcpOptionsId']
+    except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
+        module.fail_json_aws(e, msg="Unable to create dhcp option set")
 
+    return changed, None
 
-def remove_dhcp_options_by_id(vpc_conn, dhcp_options_id):
-    associations = vpc_conn.get_all_vpcs(filters={'dhcpOptionsId': dhcp_options_id})
-    if len(associations) > 0:
-        return False
+
+def find_opt_index(config, option):
+    # TODO: I don't like this but I guess it works?
+    return (next((i for i, item in enumerate(config) if item["Key"] == option), None))
+
+
+def inherit_dhcp_config(existing_config, new_config):
+    # Compare two DhcpConfigurations lists and apply existing options to unset parameters
+    """
+    If there's an existing option and the new option is not an empty list, and
+    both, the new option is not set or it's none
+    set new to existing
+    """
+    changed = False
+    for option in ['domain-name', 'domain-name-servers', 'ntp-servers',
+                   'netbios-name-servers', 'netbios-node-type']:
+        existing_index = find_opt_index(existing_config, option)
+        new_index = find_opt_index(new_config, option)
+        # if existing_index and new_index:
+        #     if existing_config[existing_index]['Value'] != new_config[new_index]['Value']:
+        #         new_config[new_index]['Value'] = existing_config[existing_index]['Value']
+        if existing_config and not new_index:
+            new_config.append(existing_config[existing_index])
+            changed = True
+
+    return changed, new_config
+
+
+def normalize_config(option_config):
```

Comment:
This seems to be a good candidate for an unit-test.

Reply:
I'm still working through all these data transformations (I keep finding cases where they don't quite work) but you're absolutely right, they should all be covered.